sample/lib/src/commonMain/kotlin/software/amazon/lastmile/kotlin/inject/anvil/sample/AssistedIdProvider.kt

```diff
@@ -0,0 +1,26 @@
+package software.amazon.lastmile.kotlin.inject.anvil.sample
+
+import me.tatarka.inject.annotations.Assisted
+import me.tatarka.inject.annotations.Inject
+import software.amazon.lastmile.kotlin.inject.anvil.AppScope
+import software.amazon.lastmile.kotlin.inject.anvil.ContributesBinding
+import software.amazon.lastmile.kotlin.inject.anvil.SingleIn
+
+/**
+ * API to get the application ID. This is used in assisted injection scenarios to provide the
+ * application ID to the object graph with an extra parameter provided at runtime.
+ */
+interface AssistedApplicationIdProvider {
+    val applicationId: String
+}
+
+@Inject
+@ContributesBinding(AppScope::class, boundType = AssistedApplicationIdProvider::class)
+@SingleIn(AppScope::class)
+class RealAssistedIdProvider(
+    @Assisted private val prependedName: String,
```

Comment:
The generated code from this will now generate the following component: 

```kotlin
@Origin(value = RealAssistedIdProvider::class)
public interface SoftwareAmazonLastmileKotlinInjectAnvilSampleRealAssistedIdProvider {
  @Provides
  public fun provideRealAssistedIdProviderAssistedApplicationIdProvider(@Assisted prependedName: String, applicationIdProvider: ApplicationIdProvider): AssistedApplicationIdProvider = RealAssistedIdProvider(prependedName, applicationIdProvider)
}
```